lib/mayaUsd/utils/util.cpp

```diff
@@ -2101,3 +2133,48 @@ UsdMayaUtil::nameToDagPath(const std::string& name)
     CHECK_MSTATUS(status);
     return dag;
 }
+
+void 
+UsdMayaUtil::GetFilteredSelectionToExport(bool exportSelected, MSelectionList& objectList, UsdMayaUtil::MDagPathSet& dagPaths)
```

Comment:
Had hoped that there was an easy API call or Maya command to do what I want here but I never found anything.  Most of the code I found that did something similar ended up traversing the entire Maya scene using various dag iterators, sometimes multiple times, and it didn’t fit what we want to do here very well.

What I ended up doing was:
1. Gather input in a MSelectionList
2. Move the selection list into a set, sorted by hierarchy depth.
3. Iterate over that set, so kind of depth first traversal of the input, not the entire Maya scene, moving objects we want to keep into a second set that will be returned to the caller.